// Protocol buffer for the cache configuration
// By: Nick from CoffeeBeforeArch
// 4/15/2020

syntax = "proto3";

// Configuration settings for each CacheLevel
message CacheConfig {
  // Size of the cache (in bytes)
  uint64 cache_size = 1;

  // Size of a cache line (in bytes)
  uint32 line_size = 2;

  // Associativity (number of ways)
  uint32 associativity = 3;
  
  // Miss penalty (in cycles)
  uint32 miss_penalty = 4;

  // Replacement policy (e.g., LRU)
  string replacement_policy = 5;

  // Write policy (e.g., WriteThrough)
  string write_policy = 6;

  // Write miss policy (e.g., Allocate)
  string write_miss_policy = 7;
  
  // Trace file location
  string trace_file_path = 8;
}
